app/models/concerns/hostext/search.rb

```diff
@@ -65,7 +65,7 @@ module Search
 
       scoped_search :relation => :puppetclasses, :on => :name, :complete_value => true, :rename => :class, :only_explicit => true, :operators => ['= ', '~ '], :ext_method => :search_by_puppetclass
       scoped_search :relation => :fact_values, :on => :value, :in_key => :fact_names, :on_key => :name, :rename => :facts, :complete_value => true, :only_explicit => true, :ext_method => :search_cast_facts
-      scoped_search :relation => :search_parameters, :on => :value, :on_key => :name, :complete_value => true, :rename => :params, :ext_method => :search_by_params, :only_explicit => true
```

Comment:
I know, I'm late to the party. But where did this go?

Reply:
As value field is now converted to serialized one searching is not working as expected.
I tried using `to_yaml` but won't able to cover all scenarios and for smart_class_parameters, I don't find searching with value field. 
One of suggestion given by @ekohl is to have searchable field separately and value with yaml. 
After this [discussion](https://github.com/theforeman/foreman/pull/5241#issuecomment-392332053), I have removed searching parameters using value field and added searching based on name.
What is your thoughts on this?

Reply:
I believe we (dm) use search by parameters in some automation via API requests. We (foreman) promise that the APIv2 is stable and I believe part of the promise is keeping the search stable.

I think the least we should do is add this to the manual in the upgrade notes.

Reply:
can't we keep the current behavior ? this will continue to work with strings as it was prior to the change?

Reply:
I think the search syntax can't be considered as part of the API. While we should try to do as little breaking changes as possible, but sometimes it does not make sense. From time to time we realize it's better to rename the search keyword for a given attribute or change the :only_explicit definition or drop some operators for some attribute and it was always accepted.

Now for the suggestion, if we allow fulltext searching, it could work in some cases. But it can also start returning weird results or simply feel buggy. People will ask for adding more operators in order to search in nested keys etc. I think it's better to just drop this and don't make people confused. I believe it's clear to all devs, how the search would behave, but it would be quite surprising for users who don't understand how we store data in SQL and how scoped search syntax translates to it.

Also given it doesn't work on smart class parameters and smart variables, I don't think it will hit majority of users. If you strongly believe we should add full text searching back, I'd prefer if we first add some inline help about how the search works and especially in this particular search field.

Reply:
> I think the search syntax can't be considered as part of the API.

I think the search syntax is a valid part of the API, we even use it for permissions. As I said, the least we should do is document the changes so users don't get surprises when they upgrade.

Reply:
So here's the documentation - https://github.com/theforeman/theforeman.org/pull/1322
Given the fact ale least two people feel differently about that, we should probably bring that up to discourse or document the outcome in our manual. At the end I don't mind steering any direction, but we should set the expectations for users. I didn't find any mention at https://theforeman.org/manuals/1.20/index.html#4.1.5Searching (or in API section linked from here)